### DISCUSSION TITLE
32-bit JDK for Windows x86?

### DISCUSSION THREAD

**Discussion by balgillo** [microsoft/openjdk#154] [Q&A] [answered]
I can only see 64 bit JDKs for download at https://docs.microsoft.com/en-gb/java/openjdk/download

Will there be a 32 bit JDK for Windows?

This would allow us to move over from AdoptOpenJDK, which [does provide x86 Windows builds](https://adoptium.net/releases.html?variant=openjdk11).

**Comment by iskimos** [comment]
Please keep in mind another one option https://bell-sw.com/pages/downloads here builds for x86_32 in place

**Comment by karianna** [comment, marked as answer]
Hi @balgillo - we are not currently planning on releasing x86-32 builds for Windows, but I'd like to hear more about your use case if you can elaborate here a little more?

**Reply by balgillo** [reply]
We have a legacy application that uses a plethora of 32 bit DLLs via JNI.  Ideally we'll move it to 64 bit and replace all the DLLs, but some may be difficult (e.g. [JDIC](https://stackoverflow.com/questions/4641698/what-files-does-jdic-need-to-run)).  If there were a Microsoft 32 bit JDK, we could move to that without having to do this big exercise of moving to 64 bit as a prerequisite.

**Reply by karianna** [reply]
That is a challenge! Unfortunately, our general policy is not to produce MSFT Builds of OpenJDK for O/S's that themselves are unsupported (including Windows 7 which is now out of support).  Probably worth a deeper discussion though if you can drop me an email to openjdk-support AT Microsoft DOT com

**Reply by balgillo** [reply]
Thanks, I appreciate the answer.  It's not to support 32-bit Windows, but to run as a 32-bit process on 64-bit Windows, so still a supported OS.

:eyes: The "competitors" AdoptOpenJDK and Liberica JDK do offer an x86 Windows build.

I'll follow up by email with more details of our use case.

**Reply by balgillo** [reply]
Another point: as far as I can tell, 32-bit Windows 10 is still supported, it's just not offered to OEMs.  That info comes from this article: https://www.engadget.com/windows-10-32-bit-oem-173055990.html

**Reply by karianna** [reply]
Thanks for the clarifications!
